### PR TITLE
PSR-0 compatibility

### PR DESCRIPTION
**Pull request by jaimeperez**
This PR adds basic PSR-0 compatibility as discussed in issue #14. I've just tested and autoloading continues to work perfectly fine. All the classes kept their original names, so there's no changes in the API. I've bumped the version to 1.4.0 in each file, and taken the opportunity to reformat slightly the code to comply also with PSR coding standards.

Next step, as suggested in #14, would be to start supporting namespaces. That would break the current API so I would do it in steps, bumping the version to 2.0, but I'm also fine if you just want to label this 2.0 and start using namespaces straight away. I can add a commit with namespaces to this PR if we agree on that.

My initial suggestion would be:

* `XMLSecurity\Key`: src/Key.php
* `XMLSecurity\DSig`: src/DSig.php
* `XMLSecurity\Encryption`: src/Encryption.php

### CODE REVIEW

**Comment by jaimeperez** [comment]
Actually, it'd be better if we directly try to go for PSR-4, as PSR-0 is already deprecated. Then we could have:

* `\RobRichards\XMLSecurity\Key`: XMLSecurity/Key.php
* `\RobRichards\XMLSecurity\DSig`: XMLSecurity/DSig.php
* `\RobRichards\XMLSecurity\Encryption`: XMLSecurity/Encryption.php

File: composer.json [hunk 1]
```diff
@@ -10,7 +10,14 @@
   ],
   "homepage": "https://github.com/robrichards/xmlseclibs",
   "autoload": {
-    "files": ["xmlseclibs.php"]
+    "psr-0": {
+      "": "src/"
+    }
+  },
+  "autoload-dev": {
```
Comment (Maks3w) [inline composer.json]: Remove this feature as is not used by the project.

File: src/XMLSecEnc.php [hunk 2]
```diff
@@ -0,0 +1,436 @@
+<?php
+
+/**
+ * Class to handle XML encryption.
+ *
+ * @author     Robert Richards <rrichards@cdatazone.org>
+ * @copyright  2007-2015 Robert Richards <rrichards@cdatazone.org>
+ * @license    http://www.opensource.org/licenses/bsd-license.php  BSD License
+ * @version    1.4.0
```
Comment (Maks3w) [inline src/XMLSecEnc.php]: I would like to remove `@version` tags from source code

**Comment by Maks3w** [comment]
The PR is incomplete. Tests are broken. `require(dirname(__FILE__) . '/../xmlseclibs.php');` must be replaced with composer autoloader

**Comment by robrichards** [comment]
@Maks3w @jaimeperez We need to leave the version number in and I can't have it dependent upon composer or an autoloader. I have some known use cases which I need to continue to support.

**Comment by Maks3w** [comment]
@robrichards namespaces and PSR-4 is a must do. Development is very painful with the monolithic file.

What I suggest then is to concat individual files in a generated one when release if you need to keep only one file.

**Comment by robrichards** [comment]
@Maks3w not looking at keeping one monolithic file as modifying the directory structure is fine. I just am planning on modifying the xmlseclibs file so it can be used similar to how it used today. I need to make sure that as we go down this path we are not focusing strictly on the composer and similar cases and ignoring the other use cases which I still need to support

**Comment by Maks3w** [comment]
You could have a single file with require_once to the rest of files

**Comment by jaimeperez** [comment]
On 28 May 2015, at 02:30, Maks3w <notifications@github.com<mailto:notifications@github.com>> wrote:


The PR is incomplete. Tests are broken. require(dirname(__FILE__) . '/../xmlseclibs.php'); must be replaced with composer autoloader

I've never considered this PR complete. I think such change needs quite a lot of thinking and discussion before we can actually merge it, so I'd consider this more of a demonstration of how it could work...

**Comment by jaimeperez** [comment]
On 28 May 2015, at 17:21, Maks3w <notifications@github.com<mailto:notifications@github.com>> wrote:


You could have a single file with require_once to the rest of files

Agreed.

Rob, the use cases you need to support... would those keep you from bumping the supported PHP version? Or is it just that you need to provide a file to include?

**Comment by robrichards** [comment]
It ended up easier to just make the changes manually as this went into the 1.4 branch and we can start clean against master for namespace support. Otherwise change is there now

**Pull request closed without merging**